src/database/DataTypes/StocktakeItem.js

```diff
@@ -243,6 +243,76 @@ export class StocktakeItem extends Realm.Object {
     const itemBatch = createRecord(database, 'ItemBatch', this.item, batchString);
     createRecord(database, 'StocktakeBatch', this, itemBatch, true);
   }
+
+  /**
+   * Applies the given Options object to all stocktake batches assosciated to
+   * this stocktake item, if there is a difference between countedTotalQuantity
+   * and snapshotTotalQuantity.
+   * @param {Realm}   database
+   * @param {Options} option
+   */
+  applyReasonToBatches(database, option) {
+    this.batches.forEach(batch => {
+      if (batch.countedTotalQuantity !== batch.snapshotTotalQuantity) {
+        database.write(() => {
+          database.update('StocktakeBatch', { id: batch.id, option });
+        });
+      }
+    });
+  }
+
+  /**
+   * Removes all Options objects from the stocktake batches assosciated
+   * with this stocktake item.
+   * @param {Realm} database
+   */
+  removeReason(database) {
+    this.batches.forEach(batch => {
+      if (batch.option) {
+        database.write(() => {
+          database.update('StocktakeBatch', { ...batch, option: null });
+        });
+      }
+    });
+  }
+
+  /**
+   * Finds the mode option within this stocktake items batches
+   * @return {string} The title of the reason with the highest frequency
+   */
+  get mostUsedReason() {
+    if (!this.batches) return false;
+    if (!this.batches[0]) return false;
+
+    const optionsCounts = {};
+    const options = {};
+    this.batches.forEach(batch => {
+      const { option } = batch;
+      if (option) {
+        optionsCounts[option.id] = optionsCounts[option.id] ? optionsCounts[option.id] + 1 : 1;
```

Comment:
If the ID of the `Options` object exists in the `optionsCounts` object, increment by one. Otherwise, this ID has not had a count started, so initialize to 1. Not seeing what you're seeing?

Reply:
I see, me dum dum that's fine.